### PR TITLE
Add Test Case

### PR DESCRIPTION
**Pull request by Tomoko-hjf**
### PR Docs
Others

### PR APIs
Others

### Description
Add test case

### CODE REVIEW

File: paconvert/api_matcher.py [hunk 1]
```diff
@@ -3339,3 +3339,11 @@ def generate_code(self, kwargs):
         )
 
         return code
+
+
+class InplaceMatcher(BaseMatcher):
+    def generate_code(self, kwargs):
+
+        if "inplace" in kwargs and kwargs["inplace"] != "(False)":
```
Comment (zhwesky2010) [inline paconvert/api_matcher.py:3347]: 这个就按去除inplace参数来转，我理解除了多占用一点显存没有其他的影响，这里有什么其他问题吗

**Comment by zhwesky2010** [comment]
CI需要修复一下

File: paconvert/api_matcher.py (hunk 1 above)
Comment (Tomoko-hjf) [reply paconvert/api_matcher.py]: 好的，已修改~

**Comment by Tomoko-hjf** [comment]
已按照API问题建议进行了修改
* 增加`TupleAssignMatcher`处理元组赋值问题
* `torch.diff` 在 `n != 1`时会返回`None`
* `torch.fliplr` 的`axis=1`可以处理任意维度的输入
* 已删除`torch.vdot`的`matcher`
* `paddle.einsum`确实存在这个`API`，但是表达式使用上好像存在一些差异，会再确认一下并在下个PR进行修改

File: paconvert/api_mapping.json [hunk 2]
```diff
@@ -3161,7 +3161,6 @@
     }
   },
   "torch.vdot": {
-    "Matcher": "GenericMatcher",
     "paddle_api": "paddle.dot",
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:3164]: 这个paddle_api也直接都删除吧，因为看对不上

File: paconvert/api_mapping.json [hunk 3]
```diff
@@ -3989,11 +4005,11 @@
     "paddle_api": "paddle.nn.Flatten",
     "args_list": [
       "start_dim",
-      "stop_dim"
+      "end_dim"
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:4008]: 对了，修改单测时发现映射关系需要调整，映射文档也需要相应的进行同步修改

文档也提个PR吧，包括之前的PR，放到描述里

File: tests/test_min.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        import torch
+        x = torch.tensor([[1, 2, 1], [3, 4, 6]])
+        out = [torch.tensor(0), torch.tensor(1)]
+        torch.min(x, dim=1, keepdim=False, out=out)
+        """
+    )
+    obj.run(pytorch_code, ["out"])
+
+
+def _test_case_7():
```
Comment (zhwesky2010) [inline tests/test_min.py:88]: int单测case可以少写一些，因为使用场景比较少，大部分还是float的

这里不能跑，都是因为不支持int原因吗

File: tests/test_nn_AdaptiveMaxPool1d.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        x = torch.tensor([[[-1.3020, -0.1005,  0.5766,  0.6351, -0.8893,  0.0253, -0.1756, 1.2913],
+                            [-0.8833, -0.1369, -0.0168, -0.5409, -0.1511, -0.1240, -1.1870, -1.8816]]])
+        model = nn.AdaptiveMaxPool1d(5, False)
+        result = model(x)
+        """
+    )
+    obj.run(pytorch_code, ["result"])
+
+
+def _test_case_4():
```
Comment (zhwesky2010) [inline tests/test_nn_AdaptiveMaxPool1d.py:64]: 以后这里的单测注释掉，都备注一下原因吧，方便后期追踪修复

以前的也需要备注下

File: tests/test_min.py (hunk 4 above)
Comment (Tomoko-hjf) [reply tests/test_min.py:88]: 好的，int的case会少写一些，这里不通过的原因是指定dim后，pytorch会返回索引下标，不是int的问题。

File: tests/test_nn_AdaptiveMaxPool1d.py (hunk 5 above)
Comment (Tomoko-hjf) [reply tests/test_nn_AdaptiveMaxPool1d.py:64]: 好的，下个PR我统一把之前的都加上吧

File: paconvert/api_mapping.json (hunk 3 above)
Comment (Tomoko-hjf) [reply paconvert/api_mapping.json:4007]: 好的

File: paconvert/api_mapping.json (hunk 2 above)
Comment (Tomoko-hjf) [reply paconvert/api_mapping.json]: done

File: paconvert/api_mapping.json (hunk 3 above)
Comment (Tomoko-hjf) [reply paconvert/api_mapping.json:4007]: 这里的映射关系看起来是对的，下个PR会把所有已有单测的API的映射关系文档检查更新一下
image

File: tests/test_nn_AdaptiveMaxPool1d.py (hunk 5 above)
Comment (Tomoko-hjf) [reply tests/test_nn_AdaptiveMaxPool1d.py:65]: > 以后这里的单测注释掉，都备注一下原因吧，方便后期追踪修复
> 
> 以前的也需要备注下

已为所有不通过的单测加上了备注 @zhwesky2010

**Comment by zhwesky2010** [comment]
@Tomoko-hjf 有冲突了需要修复一下

**Comment by Tomoko-hjf** [comment]
> @Tomoko-hjf 有冲突了需要修复一下

已修复

File: tests/test_chain_matmul.py [hunk 6]
```diff
@@ -54,6 +54,7 @@ def test_case_3():
     obj.run(pytorch_code, ["result", "out"])
 
 
+# paddle does not support tuple or list input
 def _test_case_4():
```
Comment (zhwesky2010) [inline tests/test_chain_matmul.py:58]: 这个是不支持tuple/list 矩阵吗，看torch传入的也不是list/tuple

File: tests/test_arange.py [hunk 7]
```diff
@@ -59,6 +59,7 @@ def test_case_4():
     obj.run(pytorch_code, ["result"])
 
 
+# paddle step argument does not support float numbers
```
Comment (zhwesky2010) [inline tests/test_arange.py:62]: 这个arange的float case修好了，可以打开了，CI上默认使用最新的develop版本paddle

File: tests/test_dstack.py [hunk 8]
```diff
@@ -19,6 +19,7 @@
 obj = APIBase("torch.dstack")
 
 
+# torch will reshape 1-D and 2-D inputs into 3-D, and then operate along the third axis. paddle directly using stack, which is inconsistent in implementation.
```
Comment (zhwesky2010) [inline tests/test_dstack.py:22]: 这个有其他办法实现吗

File: tests/test_dstack.py [hunk 9]
```diff
@@ -19,6 +19,7 @@
...
 def _test_case_1():
```
Comment (zhwesky2010) [inline tests/test_dstack.py:23]: 这个test_dstack都不能跑了，如果认为Matcher有问题，就删了或重新实现

File: tests/test_einsum.py [hunk 10]
```diff
@@ -24,7 +24,7 @@ def _test_case_1():
         """
         import torch
         x = torch.tensor([[1, 2, 3],[6, 2, 9], [1, 2, 3]])
-        result = torch.einsum('ii->i', x)
+        result = torch.einsum('ij->', x)
```
Comment (zhwesky2010) [inline tests/test_einsum.py:27]: 这个case是什么问题

File: tests/test_floor_divide.py [hunk 11]
```diff
@@ -20,38 +20,40 @@
...
         """
         import torch
-        result = torch.floor_divide(input=torch.tensor([4.0, 3.0]), other=2)
+        result = torch.floor_divide(input=torch.tensor([4, 3]), other=2)
         """
     )
     obj.run(pytorch_code, ["result"])
 
 
+# paddle not support float input
```
Comment (zhwesky2010) [inline tests/test_floor_divide.py:56]: 这个已经修好了可以打开了

File: tests/test_linalg_lstsq.py [hunk 12]
```diff
@@ -19,6 +19,7 @@
 obj = APIBase("torch.linalg.lstsq")
 
 
+# The third return value's type of torch is int64 and the third return value's type of paddle is 'int32'.
```
Comment (zhwesky2010) [inline tests/test_linalg_lstsq.py:22]: 这个能绕开吗

File: tests/test_masked_select.py [hunk 13]
```diff
@@ -54,6 +54,7 @@ def test_case_3():
     obj.run(pytorch_code, ["result", "out"])
 
 
+# The paddle mask parameter does not support broadcasting and must have same shape with input.
```
Comment (zhwesky2010) [inline tests/test_masked_select.py:57]: 这个记录到了API问题了吧

File: tests/test_nn_AdaptiveMaxPool3d.py [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+                            [-1.5187,  0.0235]]]]])
+        model = nn.AdaptiveMaxPool3d(1, False)
+        result = model(x)
+        """
+    )
+    obj.run(pytorch_code, ["result"])
+
+
+# The second return value's type of torch is int64 and the second return value's type of paddle is int32.
+def _test_case_4():
```
Comment (zhwesky2010) [inline tests/test_nn_AdaptiveMaxPool3d.py:71]: 这些都记录到API diff问题了吧

File: tests/test_nn_MaxUnpool1d.py [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+            return False
+        if str(pytorch_result.dtype)[6:] != str(paddle_result.dtype)[7:]:
+            return False
+        return True
+
+
+obj = API("torch.nn.MaxUnpool1d")
+
+
+def _test_case_1():
```
Comment (zhwesky2010) [inline tests/test_nn_MaxUnpool1d.py:35]: 这个什么原因

File: tests/test_nn_LogSoftmax.py [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+                        [5.0, 6.0, 7.0, 8.0],
+                        [6.0, 7.0, 8.0, 9.0]]])
+        model = nn.LogSoftmax(dim=1)
+        result = model(x)
+        """
+    )
+    obj.run(pytorch_code, ["result"])
+
+
+def _test_case_3():
```
Comment (zhwesky2010) [inline tests/test_nn_LogSoftmax.py:58]: 这个什么原因

File: tests/test_nn_MaxUnpool1d.py [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        x = torch.tensor([[[2., 4., 6., 8.]]])
+        indices = torch.tensor([[[1, 3, 5, 7]]])
+        model = nn.MaxUnpool1d(2, 2, 0)
+        result = model(x, indices)
+        """
+    )
+    obj.run(pytorch_code, ["result"])
+
+
+def _test_case_2():
```
Comment (zhwesky2010) [inline tests/test_nn_MaxUnpool1d.py:49]: 这个什么原因

File: tests/test_nn_ModuleList.py [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+# limitations under the License.
+
+import textwrap
+
+from apibase import APIBase
+
+obj = APIBase("torch.nn.ModuleList")
+
+
+def _test_case_1():
```
Comment (zhwesky2010) [inline tests/test_nn_ModuleList.py:22]: 这个什么原因

File: tests/test_nn_Softmax.py [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+                        [5.0, 6.0, 7.0, 8.0],
+                        [6.0, 7.0, 8.0, 9.0]]])
+        model = nn.Softmax(dim=1)
+        result = model(x)
+        """
+    )
+    obj.run(pytorch_code, ["result"])
+
+
+def _test_case_3():
```
Comment (zhwesky2010) [inline tests/test_nn_Softmax.py:58]: 这个什么原因

**Comment by zhwesky2010** [comment]
对了，现在0D的问题也修好了，在默认检查逻辑 check函数里也加下 shape的检查逻辑吧

File: tests/test_chain_matmul.py (hunk 6 above)
Comment (Tomoko-hjf) [reply tests/test_chain_matmul.py:58]: torch的输入是可变参数，考虑到如果输入是`*matrixs`这种情况，这种情况下我们不能用连乘实现吧（因为不知道里面有哪些元素），已经记录到api问题中。
```python
import torch
v = torch.tensor([[3., 6, 9], [1, 3, 5], [2, 2, 2]])
matrixs = (v, v, v)
result = torch.chain_matmul(*matrixs)
```

File: paconvert/api_mapping.json [hunk 20]
```diff
@@ -3705,10 +3723,7 @@
       "init",
       "device",
       "dtype"
-    ],
-    "kwargs_change": {
-      "dtype": ""
```
[the remark this answers is not in the document]
Comment (Tomoko-hjf) [reply paconvert/api_mapping.json]: done

File: paconvert/api_matcher.py [hunk 21]
```diff
@@ -3542,3 +3542,51 @@ def generate_code(self, kwargs):
...
+            code = API_TEMPLATE.format(
+                self.get_paddle_api(), self.kwargs_to_str(kwargs), out_v, out_v
+            )
+            return code.strip("\n")
+        else:
+            code = "{}({})".format(self.get_paddle_api(), self.kwargs_to_str(kwargs))
+            return code.strip("\n")
+
+
+class FloorDivideMatcher(BaseMatcher):
```
[the remark this answers is not in the document]
Comment (Tomoko-hjf) [reply paconvert/api_matcher.py]: done

File: tests/test_arange.py (hunk 7 above)
Comment (Tomoko-hjf) [reply tests/test_arange.py]: done

File: tests/test_dstack.py (hunk 9 above)
Comment (Tomoko-hjf) [reply tests/test_dstack.py]: 先从这次PR中移除

File: tests/test_floor_divide.py (hunk 11 above)
Comment (Tomoko-hjf) [reply tests/test_floor_divide.py:55]: 先从这次PR拆出来

File: tests/test_nn_AdaptiveMaxPool3d.py (hunk 14 above)
Comment (Tomoko-hjf) [reply tests/test_nn_AdaptiveMaxPool3d.py:71]: done

File: tests/test_nn_LogSoftmax.py (hunk 16 above)
Comment (Tomoko-hjf) [reply tests/test_nn_LogSoftmax.py]: 先从这次PR拆出来

File: tests/test_nn_MaxUnpool1d.py (hunk 17 above)
Comment (Tomoko-hjf) [reply tests/test_nn_MaxUnpool1d.py]: 先从这次PR拆出来

File: tests/test_nn_ModuleList.py (hunk 18 above)
Comment (Tomoko-hjf) [reply tests/test_nn_ModuleList.py]: 先从这次PR拆出来

File: tests/test_nn_Softmax.py (hunk 19 above)
Comment (Tomoko-hjf) [reply tests/test_nn_Softmax.py]: 先从这次PR拆出来

**Comment by Tomoko-hjf** [comment]
> 对了，现在0D的问题也修好了，在默认检查逻辑 check函数里也加下 shape的检查逻辑吧

修改如下后单测会报错：

**Comment by zhwesky2010** [comment]
现在0D的shape刚设置好了，如果有问题，就针对这个单测关掉shape检查吧。默认打开

另外 被禁掉的case需 迁移到新增的 **单测检查不支持** ，不能再直接禁用了

**Comment by zhwesky2010** [comment]
PR粒度小一些，以后一个PR不要超过40个文件

**Comment by zhwesky2010** [comment]
改动Matcher时，对应的映射文档如果需要修改的话也要同步修改，不要Matcher的bug修了，文档的bug一直都在
